exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/HashJoinPrel.java

```diff
@@ -50,14 +51,21 @@
   private int joinControl;
 
   public HashJoinPrel(RelOptCluster cluster, RelTraitSet traits, RelNode left, RelNode right, RexNode condition,
-                      JoinRelType joinType) throws InvalidRelException {
-    this(cluster, traits, left, right, condition, joinType, false, null, false, JoinControl.DEFAULT);
+                      JoinRelType joinType, boolean semiJoin) throws InvalidRelException {
+    this(cluster, traits, left, right, condition, joinType, false, null, false, JoinControl.DEFAULT, semiJoin);
+  }
+
+  public HashJoinPrel(RelOptCluster cluster, RelTraitSet traits, RelNode left, RelNode right, RexNode condition,
+                      JoinRelType joinType, boolean swapped, RuntimeFilterDef runtimeFilterDef,
+                      boolean isRowKeyJoin, int joinControl) throws InvalidRelException {
+    this(cluster, traits, left, right, condition, joinType, swapped, runtimeFilterDef, isRowKeyJoin, joinControl, false);
   }
 
   public HashJoinPrel(RelOptCluster cluster, RelTraitSet traits, RelNode left, RelNode right, RexNode condition,
       JoinRelType joinType, boolean swapped, RuntimeFilterDef runtimeFilterDef,
-      boolean isRowKeyJoin, int joinControl) throws InvalidRelException {
-    super(cluster, traits, left, right, condition, joinType);
+      boolean isRowKeyJoin, int joinControl, boolean semiJoin) throws InvalidRelException {
+    super(cluster, traits, left, right, condition, joinType, semiJoin);
+    Preconditions.checkArgument(isSemiJoin && !swapped || swapped && !isSemiJoin || (!swapped && !isSemiJoin));
```

Comment:
Why not simplify to :  
`if (isSemiJoin) {`
   `Preconditions.checkArgument(!swapped : 'swapping of inputs is not allowed for semi-joins') `
`}`